test/FadeMixinSpec.js

```diff
@@ -54,4 +55,23 @@ describe('FadeMixin', function () {
       }, 25);
     }, 25);
   });
+
+  it('should retain the same input value throughout transition', function (done) {
+    let instance = ReactTestUtils.renderIntoDocument(<Component />);
+    let textarea = ReactTestUtils.findRenderedDOMComponentWithTag(instance, 'textarea');
+
+    setTimeout(function(){
+
+      React.findDOMNode(textarea).value = 'b';
+
+      instance.componentWillUnmount();
+      let element = instance._fadeOutEl.children[0];
+      let child = element.children[1];
```

Comment:
This is so cryptic, unreadable and brittle :sweat:

Could you please re-write the test such it is all clear what's going on and what we are testing here ?
:cherries: